jax/_src/lax/lax.py

```diff
@@ -966,12 +966,45 @@ class GatherDimensionNumbers(NamedTuple):
   start_index_map: Sequence[int]
 
 
+class GatherScatterMode(IntEnum):
```

Comment:
Is there any particular reason for using `IntEnum` here? Generally `Enum` is a better choice for non-legacy code, which does not need to support integer values.

Reply:
Changed to `Enum`. Thanks for the tip!